doc/api/stream.md

```diff
@@ -1555,12 +1555,18 @@ const cleanup = finished(rs, (err) => {
 });
 ```
 
-### `stream.pipeline(...streams, callback)`
+### `stream.pipeline(src, ...transforms, dst, callback)`
 <!-- YAML
 added: v10.0.0
 -->
 
-* `...streams` {Stream} Two or more streams to pipe between.
+* `src` {Readable|Iterable|AsyncIterable|Function|AsyncFunction}
+  returns an `Iterable` or `AsyncIterable`.
+* `...transforms` {Duplex|Function|AsyncFunciton}
+  returns an `Iterable` or `AsyncIterable`.
+  * `source` {AsyncIterable}
+* `dst` {Writable|Function}.
+  * `source` {AsyncIterable}
 * `callback` {Function} Called when the pipeline is fully done.
```

Comment:
I need some help with the above. The arguments here become rather complicated. @nodejs/documentation 

`src` can be a Readable, Iterable, AsyncIterable or function returning a Iterable, AsyncIterable, Promise<Iterable> or Promise<AsyncIterable>.

`dst` argument must be a Writable or an async function taking an AsyncIterable.

`...transforms` can either be a Duplex or function taking an AsyncIterable, and returning a AsyncIterable, Promise<Iterable> or Promise<AsyncIterable>.

Reply:
what I would recommend is separating these out into separate sections... that is, have one `### stream.pipeline(...` for each significant variation.

Reply:
@jasnell: That becomes quite a lot of permutation though. Or am I misunderstanding?